packages/back-end/src/api/sdk-connections/validations.ts

```diff
@@ -0,0 +1,116 @@
+import { getLatestSDKVersion, getSDKCapabilities } from "shared/sdk-versioning";
+import { findAllProjectsByOrganization } from "../../models/ProjectModel";
+import { ApiReqContext } from "../../../types/api";
+import { sdkLanguages } from "../../util/constants";
+import { getEnvironments } from "../../services/organizations";
+
+const capabilityParams = [
+  ["encryption", "encryptPayload"],
+  ["remoteEval", "remoteEvalEnabled"],
+] as const;
+
+type CababilitiesParamKey = typeof capabilityParams[number][1];
+type CababilitiesParams = { [k in CababilitiesParamKey]?: boolean };
+
+const premiumFeatures = [
+  ["encrypt-features-endpoint", "encryptPayload"],
```

Comment:
Also add `visual-editor`, `hash-secure-attributes`, `remote-evaluation`, and `redirects`.

In addition, if the user doesn't have `cloud-proxy` and IS_CLOUD, then they shouldn't be able to specify any of the proxy settings.